server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/SieveScriptRoutes.java

```diff
@@ -0,0 +1,187 @@
+    public static final String SCRIPTS = "scripts";
+    private static final String USER_NAME = "userName";
+    private static final String SCRIPT_NAME = "scriptName";
+    private static final String ACTIVATE_PARAMS = "activate";
+    private static final String USER_SCRIPT_PATH = Joiner.on(SEPARATOR)
+        .join(ROOT_PATH, ":" + USER_NAME, SCRIPTS, ":" + SCRIPT_NAME);
+
+    private final SieveRepository sieveRepository;
+    private final UsersRepository usersRepository;
+
+    @Inject
+    public SieveScriptRoutes(SieveRepository sieveRepository, UsersRepository usersRepository) {
+        this.sieveRepository = sieveRepository;
+        this.usersRepository = usersRepository;
+    }
+
+    @Override
+    public String getBasePath() {
+        return ROOT_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        defineAddActiveSieveScript(service);
+    }
+
+    @PUT
+    @ApiOperation(value = "Upload a new Sieve Script")
+    @Path(value = ROOT_PATH + "/{" + USER_NAME + "}/" + SCRIPTS + "/{" + SCRIPT_NAME + "}")
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "OK"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid username"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid Sieve script name"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Empty script is not accepted"),
+        @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "User not exist")
+    })
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            name = USER_NAME,
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            example = ROOT_PATH + "/userNameA/" + SCRIPTS + "/scriptName1",
+            value = "Username"),
+        @ApiImplicitParam(
+            name = SCRIPT_NAME,
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            example = ROOT_PATH + "/userNameA/" + SCRIPTS + "/scriptName1",
+            value = "Script name"),
+        @ApiImplicitParam(
+            required = false,
+            paramType = "query parameter",
+            dataType = "Boolean",
+            defaultValue = "False",
+            example = "?activate=true",
+            value = "If present, automatically activating the script.")
+    })
+    public void defineAddActiveSieveScript(Service service) {
+        service.put(USER_SCRIPT_PATH, this::addActiveSieveScript);
+    }
+
+    private HaltException addActiveSieveScript(Request request, Response response) throws UsersRepositoryException, QuotaExceededException, StorageException, ScriptNotFoundException {
+        User user = extractUser(request);
+        ScriptName script = extractScriptName(request);
+        sieveRepository.putScript(user, script, extractSieveScriptFromRequest(request));
+        if (isActivated(request.queryParams(ACTIVATE_PARAMS))) {
```

Comment:
you already put the script before validating query params, you can return an 400 but have an effect on the system.